nemoguardrails/actions/llm/generation.py

```diff
@@ -354,10 +396,14 @@ async def generate_user_intent(
             )
         # The last event should be the "StartInternalSystemAction" and the one before it the "UtteranceUserActionFinished".
         event = get_last_user_utterance_event(events)
+        assert event
         assert event["type"] == "UserMessage"
```

Comment:
not related to your changes, but [should not use assert](https://google.github.io/styleguide/pyguide.html#244-decision:~:text=Do%20not%20use%20assert%20statements%20in%20place%20of%20conditionals%20or%20validating%20preconditions.)

```python
  if not event:
      raise ValueError(
          "No user message found in event stream. Unable to generate user intent."
      )
  if event["type"] != "UserMessage":
      raise ValueError(
          f"Expected UserMessage event, but found {event['type']}. "
          "Cannot generate user intent from this event type."
      )
```